### PR TITLE
IBL drag & drop support for desktop gltf_viewer

### PR DESCRIPTION
**Pull request by z3moon**
Users now can drag and drop IBL files for desktop gltf_viewer.

It attempts to load gltf files first then tries to load IBL files.

### CODE REVIEW

File: samples/gltf_viewer.cpp [hunk 1]
```diff
@@ -508,30 +508,64 @@ static void onClick(App& app, View* view, ImVec2 pos) {
...
     if (!filename.exists()) {
         std::cerr << "file " << filename << " not found!" << std::endl;
         return {};
     }
+
     if (filename.isDirectory()) {
-        auto files = filename.listContents();
-        for (const auto& file: files) {
-            if (file.getExtension() == "gltf" || file.getExtension() == "glb") {
+        bool found = false;
```
Comment (poweifeng) [inline samples/gltf_viewer.cpp:523]: use `std::find` here?

File: samples/gltf_viewer.cpp [hunk 2]
```diff
@@ -508,30 +508,64 @@ static void onClick(App& app, View* view, ImVec2 pos) {
...
+            return {};
+        }
+    } else if (!isIBL(filename)) {
+        return {};
+    }
+
+    return filename;
+}
+
+static utils::Path getPathForGLTFAsset(std::string_view string) {
```
Comment (poweifeng) [inline samples/gltf_viewer.cpp:540]: I feel like this and `getPathForIBLAsset` are pretty much the same except for `isGLTF` and `isIBL`.  Maybe we can pull that into a paramter and just have a single method?

**Review by pixelflinger — APPROVED** [review]
LGTM after addressing powei's comments.

File: samples/gltf_viewer.cpp (hunk 1 above)
Comment (z3moon) [reply samples/gltf_viewer.cpp]: good point. let me update it.

File: samples/gltf_viewer.cpp (hunk 2 above)
Comment (z3moon) [reply samples/gltf_viewer.cpp:535]: Yeah I actually considered unifying them at first. Though it's possible to do that, it's quite tricky to distinguish the slightly different but critical behaviors. So it's likely to end up with convoluted code.

**Review by poweifeng — APPROVED** [review]

**Pull request merged**